ui/lib/kv/addon/routes/secret/metadata.js

```diff
@@ -7,31 +7,63 @@ import Route from '@ember/routing/route';
 import { service } from '@ember/service';
 
 export default class KvSecretMetadataRoute extends Route {
```

Comment:
My brain struggled with how to refactor this since it previously relied a _lot_ on the `kv/data` model. Open to other ways of handling this! I _think_ what I landed on was the least invasive, but a fresh perspective would be helpful!

Reply:
Prior to the move-kv-to-ember-engine work, I had actually long ago added a capabilities check directly inside a kv component. KV is a unique case with coupled workflow and api endpoints complicating permission checks. All that to say, 1. I think calling the capability service here and not using the one from the model makes sense. 2. your structure of fetchCapabilities makes sense; kind of a fresh take on our more long handed approach we use now in the models. Nice work.